### ISSUE TITLE
Delete.Objects.map is not a function when trying to delete a single file via S3 DeleteObjects Command

### ISSUE THREAD

**Issue by michaeljauk** [supabase/storage#504]
# Bug report

- [x] I confirm this is a bug with Supabase, not with my own application. 
- [x] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

When trying to delete a single file via the S3 API (calling `POST /s3/<bucket>/?delete=&x-id=DeleteObjects`, DeleteObjects Command) a `HTTP 500` error gets returned and no file gets deleted. In the corresponding logs of Supbase the following error gets displayed (abbreviated for brevity):

```json
...
"error": [
  {
    "message": "Delete.Objects.map is not a function",
    "name": "TypeError",
    "raw": "{}",
    "stack": "TypeError: Delete.Objects.map is not a function\n    at S3ProtocolHandler.deleteObjects (/app/dist/storage/protocols/s3/s3-handler.js:595:43)\n    at Object.handler (/app/dist/http/routes/s3/commands/delete-object.js:59:27)\n    at Object.routeHandler (/app/dist/http/routes/s3/index.js:44:60)\n    at preHandlerCallback (/app/node_modules/fastify/lib/handleRequest.js:128:37)\n    at next (/app/node_modules/fastify/lib/hooks.js:176:7)\n    at handleResolve (/app/node_modules/fastify/lib/hooks.js:193:5)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)"
  }
],
"event": null,
"job": null,
"jodId": null,
"level": "error",
"metadata": null,
"objectPath": null,
"operation": "storage.s3.object.delete_many",
"owner": null,
"payload": null,
...
 ```

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Upload one or more files to a bucket
2. Try to delete the file via the S3 API's `DeleteObjectsCommand`

## Expected behavior
The corresponding files should be deleted

**Comment by fenos** [comment]
Hi @michaeljauk this is interesting, I can't reproduce the issue.
Can you provide more information on how you are sending the DeleteObjects Command?

**Comment by fenos** [comment]
In the PR above I've added more strict validation. 
But apparently, the command you are sending doesn't seem to be conforming with the Spec.

Once you have more information on this, I'll be happy to dig further

**Comment by fenos** [comment]
@michaeljauk thanks a lot for this! I could reproduce the issue, filing a fix shortly!

**Comment by fenos** [comment]
Ok, I have expanded the above PR and introduced the fix.
TLDR, the XML parser when it receives only an item, parses it as an object as opposed to an array.

I could force the parsing to an array by implementing the [`isArray`](https://github.com/NaturalIntelligence/fast-xml-parser/blob/master/docs/v4/2.XMLparseOptions.md#isarray) method on the fast-xml-parser

**Comment by michaeljauk** [comment]
@fenos I guess another week or 2 weeks until it is GA? 👀

**Comment by fenos** [comment]
it is now being deployed GA
